pandas/_libs/interval.pyx

```diff
@@ -413,6 +415,8 @@ cdef class Interval(IntervalMixin):
             isinstance(y, numbers.Number)
             or PyDelta_Check(y)
             or is_timedelta64_object(y)
+            or isinstance(y, _Timestamp)
+            or isinstance(y, _Timedelta)
```

Comment:
the PyDelta_Check call makes the _Timedelta check extraneous.  The remaining isinstance checks can be combined.